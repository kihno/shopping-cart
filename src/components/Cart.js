import React from 'react';

const Cart = (props) => {
    const { cart } = props;

    const totalPrice = (cart) => {
        let price = 0;

        cart.map(item => {
            price += item.price;
        });

        return price;
    }

    return(
        <div id="cart">
            {cart.map(item => {
                return <div className="cartItem" key={item.id} data-testId="cartItem">
                            <div data-testId="name">{item.name}</div>
                            <div>{item.price}</div>
                        </div>
            })}
            <div id="cartTotal" data-testId="price">{totalPrice(cart)}</div>
        </div>
    )
}

export default Cart;