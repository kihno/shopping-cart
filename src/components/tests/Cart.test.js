import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Cart from '../Cart';

const cart = [
    {name: 'foo', price: 15, id:0},
    {name: 'bar', price: 5, id:1},
    {name: 'baz', price: 3, id:2},
];

describe('Shopping cart', () => {
    it('renders cart', () => {
        const { container } = render(<Cart cart={cart} />);
        expect(container).toMatchSnapshot();
    });

    it ('renders products in cart', () => {
        render(<Cart cart={cart} />);

        expect(screen.queryAllByTestId('cartItem').length).toBe(3);
        expect(screen.queryAllByTestId('name')[0].textContent).toBe('foo');
    });

    it('renders total price', () => {
        render(<Cart cart={cart} />);

       expect(screen.queryByTestId('price').textContent).toBe('23'); 
    });
});