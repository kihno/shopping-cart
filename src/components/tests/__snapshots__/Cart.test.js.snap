// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopping cart renders cart 1`] = `
<div>
  <div
    id="cart"
  >
    <div
      class="cartItem"
      data-testid="cartItem"
    >
      <div
        data-testid="name"
      >
        foo
      </div>
      <div>
        15
      </div>
    </div>
    <div
      class="cartItem"
      data-testid="cartItem"
    >
      <div
        data-testid="name"
      >
        bar
      </div>
      <div>
        5
      </div>
    </div>
    <div
      class="cartItem"
      data-testid="cartItem"
    >
      <div
        data-testid="name"
      >
        baz
      </div>
      <div>
        3
      </div>
    </div>
    <div
      data-testid="price"
      id="cartTotal"
    >
      23
    </div>
  </div>
</div>
`;
